### PR TITLE
fix(server): report a run that executed no tests as passed

### PR DESCRIPTION
**Pull request by fortmarek**
## What changed

A test run whose xcresult parses cleanly but contains no test modules is now reported as `success` instead of `failed_processing`.

That is a one-line change in `Tuist.Tests.Workers.ProcessXcresultWorker.run_status/2`. Everything else in the diff is tests.

## Why

Nothing had failed. `xcodebuild` completed successfully, a simulator booted, the test runners installed and launched, and the selection simply resolved to zero tests. The run was labelled `failed_processing`, which reads as a Tuist side failure and sent a real support question down a long investigation.

Xcode represents this case as a **passing run**. Opening such a bundle shows green checks throughout, `Testing Complete` with `No issues`, and `0 tests total, 0 passed` per target. `xcresulttool` reports `totalTestCount: 0` and `skippedTests: 0`. Mirroring that keeps Tuist consistent both with what the developer sees in Xcode and with the green `xcodebuild` exit their CI already reported.

### Root cause of the wrong label

`failed_processing` was introduced for this case deliberately, in 399aad6a96. The reasoning was sound: the Swift parser derives an empty run's status from

```swift
testCases.allSatisfy { $0.status == .skipped }
```

which is vacuously true for an empty array, so an empty bundle surfaced as `Skipped` and was indistinguishable from a deliberate skip. `failed_processing` was the least bad existing bucket.

That traded one wrong label for another, and it overloaded a status whose meaning is that the server could not process the bundle. Those are worth separating, because genuine processing failures already have their own paths and never reach `run_status/2`: the unprocessable-input path cancels on `:bundle_invalid` and `:xcresult_not_found`, and retry exhaustion goes through `finalize_failed_processing`. Both keep `failed_processing`.

An earlier revision of this pull request added a dedicated `no_tests` status rendered neutrally. That has been reverted in favour of matching Xcode, along with everything that existed only to serve it: the status validation entry, the API schema enums, the badges, the filter options, the analytics exclusion and the CSS.

## Impact

Runs that executed no tests stop being reported as failures and count as passing, with zero test cases visible on the run.

This is a chronic class rather than an edge case. Across three weeks of production, between 5 and 10 percent of all test runs land in `failed_processing` every day, touching 20 to 80 projects daily. Of 968 such runs sampled over two days, only 2 had any test module rows, so effectively all of them are this empty-parse case rather than a real processing failure. In the same window essentially every processing job completed successfully.

No API enum changes, so clients matching on test run status are unaffected.

## The sharding tests

While tracing why the run selected nothing, a separate representation wart surfaced. `BinPacker.determine_shard_count/2` caps on unit count in every branch, so a suite plan resolving one unit gets `shard_count == 1`, which makes shard 0 double as the catch-all. The catch-all expresses "run everything not assigned elsewhere" by omission, so with nothing to subtract it passes no arguments at all and the plan's own suite assignment is discarded at read time.

Two tests characterize that, plus the asymmetry that the same suite is selected once a second unit exists.

It is deliberately not fixed here, because it is not what caused the empty run. A three shard plan for the same project and test plan two hours earlier assigned nine targets to its unfiltered catch-all; seven executed and two produced nothing, and those two are exactly the targets the collapsed single shard plan happened to contain alone. Those targets do not run under that test plan regardless of shard count, which the customer has since confirmed. Forcing an explicit selection would override a deliberate opt out.

Also deliberately untouched: the catch-all downloading products for modules with no suite history is correct and already asserted by an existing test. That is its purpose.

## How to test locally

From `server/`:

```bash
mix test test/tuist/tests_test.exs test/tuist/tests/workers/process_xcresult_worker_test.exs test/tuist/shards_test.exs
```

The behaviour is covered by `reports a bundle with no test modules as passed`. The sharding wart is covered by `collapses to a single catch-all shard that discards its own suite assignment` and `emits an explicit selection for the same suite once a second unit exists`.

Full affected surface, all passing (537 tests): `tests_test.exs`, `process_xcresult_worker_test.exs`, `shards_test.exs`, `vcs_test.exs`, `test_run_live_test.exs`, `test_runs_live_test.exs`, `tests_live_test.exs`, `shards_live_test.exs`, `tests_controller_test.exs`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/lib/tuist/tests/workers/process_xcresult_worker.ex [hunk 1]
```diff
@@ -235,12 +235,12 @@ defmodule Tuist.Tests.Workers.ProcessXcresultWorker do
...
-  # — vacuously, from an empty test-case list — which reads on the dashboard as
-  # a real skip rather than "we couldn't parse this". Surface it as
-  # failed_processing so it isn't mistaken for a passing or skipped run.
-  defp run_status(_parsed_data, []), do: "failed_processing"
+  # No test modules means xcodebuild finished without running anything: the
+  # selection resolved to zero tests. Xcode reports that as a passing run with
+  # `totalTestCount: 0` and no issues, so this mirrors it. The Swift parser's
+  # own status would be "skipped", derived vacuously from an empty test-case
+  # list, which reads as a deliberate skip instead.
+  defp run_status(_parsed_data, []), do: "success"
```
Comment (esnunes) [inline server/lib/tuist/tests/workers/process_xcresult_worker.ex:251]: Really like the reframing here — `failed_processing` for a bundle that parsed perfectly well was mixing two different meanings, and separating them is the right call.

One case I wanted to flag on this line, though: the clause matches on `test_modules == []` alone and drops `parsed_data["status"]` entirely, and there's one shape where the parser has already decided the run failed and still hands you an empty module list.

When the only thing in the bundle is a runner error — a target whose `.xctest` bundle can't be loaded, or an app/UI-test runner that can't launch — `extractErrors/4` lifts that synthetic node out of the test cases (XCResultParser.swift:317, :358), so `allTestCases` is empty and `testModules(from:)` returns `[]`. But `overallStatus` returns `.failed` in exactly that situation because `hasRunnerErrors` is true (XCResultParser.swift:288-289), which serializes as `"failure"`. So the parse arrives here as `status: "failure"`, `test_modules: []`, `errors: [the runner error]` — and this clause now persists it as `success`.

I ran that shape through `ProcessXcresultWorker.perform/1` on this branch with a stubbed parse and `create_test/1` did receive `status: "success"`. On the run page that renders a green badge sitting directly above a populated errors card, and the PR comment gets a ✅.

What makes me think this is an inversion rather than a judgement call is that the parser already draws this exact line deliberately: unattributed Swift Testing issues are specifically *not* allowed to redden a run (the comment at XCResultParser.swift:333-347 explains that xcodebuild exits 0 on them), while runner errors are routed into `.failed`.

Would keeping the parser's verdict when it has one work for you?

```elixir
defp run_status(%{"status" => "failure"}, []), do: "failure"
defp run_status(_parsed_data, []), do: "success"
```

That leaves the case this PR is actually about untouched, since its parser status is `skipped` rather than `failure`.

File: server/test/tuist/tests/workers/process_xcresult_worker_test.exs [hunk 2]
```diff
@@ -193,6 +193,33 @@ defmodule Tuist.Tests.Workers.ProcessXcresultWorkerTest do
       assert :ok == ProcessXcresultWorker.perform(oban_job(args))
     end
 
+    test "reports a bundle with no test modules as passed", %{account: account, project: project} do
```
Comment (esnunes) [inline server/test/tuist/tests/workers/process_xcresult_worker_test.exs:196]: Good that this shape is pinned — it's the case the PR is about and it's the one that was mislabelled.

The clause accepts a second input though, and nothing here reaches it: `test_modules: []` with `status: "failure"` and a non-empty `errors` list. `parsed_data_with_failure/0` always carries a populated `test_modules`, so that combination isn't covered anywhere in this file.

Whichever way the runner-error question lands, a case asserting what an empty-but-failing parse persists would make the decision explicit rather than incidental. Since `expect_local_parse/1` just returns whatever map you hand it, it's the same helper with two keys changed.

**Review by esnunes — APPROVED** [review]
This is a good call and the write-up is genuinely useful — `failed_processing` was carrying two unrelated meanings, and the fact that the two real processing-failure paths (the unprocessable-input cancel and retry exhaustion) never reach `run_status/2` is exactly the argument that makes separating them safe. I also traced the duration analytics on the way through and they don't filter on status at all, so nothing shifts in the percentiles.

One thing I'd flag before this lands, though. The new clause matches on `test_modules == []` alone and discards `parsed_data["status"]`, and there's a second shape that reaches it: a bundle whose only content is a runner error — a target whose `.xctest` bundle can't be loaded, or a UI-test runner that can't launch. The parser lifts that synthetic node out of the test cases, so `test_modules` is empty, but `overallStatus` still returns `.failed` because `hasRunnerErrors` is true. I ran that shape through the worker on this branch and it persists as `success`, with the runner error sitting in `run_errors` next to a green badge. The parser draws that line deliberately — unattributed Swift Testing issues are specifically not allowed to redden a run, runner errors are — so this reads to me like an inversion rather than a judgement call. A `defp run_status(%{"status" => "failure"}, []), do: "failure"` ahead of the new clause would leave the case you're actually targeting untouched, since its parser status is `skipped`.

One smaller note inline: the new worker test pins only the benign half of that branch.

The sharding characterization tests check out against the implementation. The touched and adjacent suites pass locally: 483 tests across `process_xcresult_worker_test.exs`, `shards_test.exs`, `tests_test.exs`, `xcresult_processing_test.exs` and `vcs_test.exs`.

File: server/lib/tuist/tests/workers/process_xcresult_worker.ex (hunk 1 above)
Comment (fortmarek) [reply server/lib/tuist/tests/workers/process_xcresult_worker.ex:251]: Confirmed and fixed in 2058a4dbed. Verified the chain you described: XCResultParser.swift:459 excludes runner-error nodes from `extractTestCases`, :289 makes `hasRunnerErrors` force `.failed`, and :268 feeds it `runnerErrors` only. Reproduced it too, and without the fix `create_test/1` received `status: "success"`.

Added `defp run_status(%{"status" => "failure"}, []), do: "failure"` ahead of the empty-module clause. Keyed on the status rather than on `errors` being non-empty, since `errors` also carries unattributed issues, which per :266/:268 deliberately leave the status alone, so an empty run with only those should still be a pass.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: server/test/tuist/tests/workers/process_xcresult_worker_test.exs (hunk 2 above)
Comment (fortmarek) [reply server/test/tuist/tests/workers/process_xcresult_worker_test.exs:196]: Added `keeps a runner error failing even though it carries no test modules` in 2058a4dbed, covering `test_modules: []` with `status: "failure"` and a populated `errors` list, asserting the run persists as `failure` with the error retained. Checked it is a real guard rather than incidental: with the new clause removed it fails with `left: "success", right: "failure"`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
